### ISSUE TITLE
Annotating images in batches and tracking progress.

### ISSUE THREAD

**Issue by Chiebukar** [SkalskiP/make-sense#236]
Since we often have to annotate lots of images its is impossible to annotate the images on one go. It will be good to be able to save current progress and continue from there later on even we leave the site or PC shuts down.

There is also no notable way to know how  many images have been annotated and the number of images remaining. This is quite useful in tracking progress of work done and left to do.

**Comment by ThangLC304** [comment]
Or you can write a simple program to move all the images with annotation into a separated folder, 
and another one to merge the new annotation file with the old (bbox annotation should be a piece of cake, polygon annotation should take you a little longer).

**Comment by SkalskiP** [comment]
Hi, @ThangLC304! 👋 Yup! The problem is that people want to use MS as a full-fledged annotation tool. At least for the time being, we have no plans to add such functionalities because their development would require a very large time investment.

**Comment by rasyidf** [comment]
We can start with project based approach. the current project name in the top center wasn't really useful.

**Comment by SkalskiP** [comment]
I have some ideas for persistence, that would not require building a backend.

1. GitHub API - The user would log in to MakeSense with his/hers GH account, and we would create per project private repo on his behalf and use it for storage. We can thal leverage [GitHub API](https://docs.github.com/en/rest/repos/repos) to read and write to that repo.  
2. The File System Access API - a new feature that is slowly rolled out in web browsers. Basically allowing websites to have access to your hard drive like it would be natively installed on your computer.

**Comment by rasyidf** [comment]
I guess [this](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryEntry ) will do .

**Comment by SkalskiP** [comment]
I'm not sure. Do you see any way to create a new file here? Also, it is quite interesting if you can create zip files this way. I was thinking we can use `.zip` as the "project" file format, and put images as well as our state JSON inside.

**Comment by rasyidf** [comment]
creating `.zip` file is a good approach. but maintaining more than 500 mb data will be much harder using zip.

my suggestion is creating project json with list of anotated file. and drop the folder that contained the json. we can crawl every object-names, get the thumbnails, then proceeds, without the need to load everything at once.

**Comment by SkalskiP** [comment]
You are right 😔 we would need to load the whole `.zip` file from the hard drive to even start processing it. Keeping images separate makes it much more sense as it allows us to load them on demand. 

The problem that I see is that then:
- One project, is stored as multiple files on the hard drive
- If a user violates the structure of the project, for example, manually deleting or moving files on disk, the entire project will be corrupted
